### PR TITLE
tests for books and categories layers

### PR DESCRIPTION
**Pull request by RedArtelerist**
image

### CODE REVIEW

File: src/test/java/maksym/fedorenko/bookstore/config/CustomMySqlContainer.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public void start() {
+        super.start();
+        System.setProperty("TEST_DB_URL", mySqlContainer.getJdbcUrl());
+        System.setProperty("TEST_DB_USERNAME", mySqlContainer.getUsername());
+        System.setProperty("TEST_DB_PASSWORD", mySqlContainer.getPassword());
+    }
+
+    @Override
+    public void stop() {
+
```
Comment (Elena-Bruyako) [inline src/test/java/maksym/fedorenko/bookstore/config/CustomMySqlContainer.java]: remove empty line

File: src/test/java/maksym/fedorenko/bookstore/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+@Sql(scripts = {
+        "classpath:database/books/delete-all-books.sql",
+        "classpath:database/categories/delete-all-categories.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static ObjectMapper objectMapper;
+
```
Comment (Elena-Bruyako) [inline src/test/java/maksym/fedorenko/bookstore/controller/BookControllerTest.java:50]: Let's add test for getAll books

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Left two comments

File: src/test/java/maksym/fedorenko/bookstore/repository/BookRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    @Autowired
+    private BookRepository bookRepository;
+
+    @BeforeAll
+    @SneakyThrows
+    static void beforeAll(@Autowired DataSource dataSource) {
+        tearDown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/repository/BookRepositoryTest.java]: in other tests you use `@Sql`

why not using it here as well?

File: src/test/java/maksym/fedorenko/bookstore/controller/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        mockMvc.perform(get("/api/categories/" + id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Find category by nonExistent id")
+    @WithMockUser(username = "user", roles = "USER")
+    void getById_WithNonExistedId_BadRequest() throws Exception {
```
Comment (NazarSmith228) [inline src/test/java/maksym/fedorenko/bookstore/controller/CategoryControllerTest.java]:
```suggestion
    void getById_WithNonExistedId_NotFound() throws Exception {
```

**Review by NazarSmith228 — APPROVED** [review]
Good job! But please pay attention to my comments

**Pull request merged**